src/Time/Actions/UpdateU.hpp

```diff
@@ -55,16 +56,14 @@ struct UpdateU {
           const auto& time_stepper =
               Parallel::get<CacheTags::TimeStepper>(cache);
 
-          history.emplace_back(time, vars, std::move(dt_vars));
+          history.emplace_back(time, vars, dt_vars);
```

Comment:
Doesn't this make things worse?  You've replaced an allocation for the DataBox with an allocation for the history and a copy into said allocation, so we seem to have no net change in allocations but an extra copy.

Reply:
Yes, if I remember correctly you and I had talked about re-using the history data at some point. I didn't want to make changes to that because I know you're still working on how the time stepper interactions with the DataBox a bit. My plan is to first not remove the dt_variables from the DataBox, then later be able to re-use the oldest entry in the history data. Is there currently a way to steal the oldest entry?

Reply:
Yes, reusing history allocations has been discussed, and still sounds like a good idea.  It shouldn't be done right now, because pull request #469 completely redoes the history structures.  (The new versions do not yet support this optimization, but I intend to add it after we get the initial version agreed upon.)  This is probably the direction UpdateU will have to go in for that, so I'm not strongly opposed to this.

I do think that the commit message needs to not claim this is saving on allocations.

Reply:
Sounds good to me. I think it would be useful for the commit message to actually explain the direction this should head. That is, make it clear there currently isn't a reduction in allocations but that this is a step in that direction and that future changes will avoid the allocation. Does that sound good to you?

Reply:
Yeah, explaining the motivation is good.

Reply:
Okay, I'll push an update. Thanks!